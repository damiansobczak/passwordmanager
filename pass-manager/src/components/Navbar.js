import React from "react";
import "./Navbar";
import { Link } from "react-router-dom";

class Navbar extends React.Component {
  render() {
    return (
      <nav className="blue accent-4">
        <div className="nav-wrapper container">
          <a href="/" className="brand-logo">
            Menadżer haseł
          </a>
          <ul id="nav-mobile" className="right hide-on-med-and-down">
            <li>
              <Link to={`/passwords`} className="nav-cart">
                Hasła
              </Link>
            </li>
            <li>
              <Link to={`/`} className="nav-cart">
                Wyloguj
              </Link>
            </li>
          </ul>
        </div>
      </nav>
    );
  }
}

export default Navbar;
