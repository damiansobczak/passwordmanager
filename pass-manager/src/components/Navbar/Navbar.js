import React from "react";
import "./Navbar.scss";
import PropTypes from "prop-types";

class Navbar extends React.Component {
  logout = e => {
    e.preventDefault();
    this.props.logout();
  };

  render() {
    return (
      <nav className="blue accent-4">
        <div className="nav-wrapper container">
          <a href="/" className="brand-logo">
            Menadżer haseł
          </a>
          <ul id="nav-mobile" className="right hide-on-med-and-down">
            {this.props.user ? (
              <li className="nav-cart">
                <a href="/" onClick={this.props.logout}>
                  {this.props.user.displayName} | Wyloguj
                </a>
              </li>
            ) : (
              ""
            )}
          </ul>
        </div>
      </nav>
    );
  }
}

Navbar.propTypes = {
  user: PropTypes.shape({
    displayName: PropTypes.string
  }),
  login: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired
};

export default Navbar;
