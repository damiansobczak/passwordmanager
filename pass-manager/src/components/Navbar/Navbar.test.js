import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Navbar from "./Navbar";

describe("<Navbar />", () => {
  it("renders correctly", () => {
    const user = {
      displayName: "Damian Sobczak"
    };
    const tree = renderer.create(<Navbar user={user} login={() => true} logout={() => true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders user name in logout section", () => {
    const user = {
      displayName: "Damian Sobczak"
    };
    const wrapper = shallow(<Navbar user={user} login={() => true} logout={() => true} />);
    expect(wrapper.find(".nav-cart a").text()).toContain(user.displayName);
  });
});
