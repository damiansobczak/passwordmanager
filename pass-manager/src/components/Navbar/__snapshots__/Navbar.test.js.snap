// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly 1`] = `
<nav
  className="blue accent-4"
>
  <div
    className="nav-wrapper container"
  >
    <a
      className="brand-logo"
      href="/"
    >
      Menadżer haseł
    </a>
    <ul
      className="right hide-on-med-and-down"
      id="nav-mobile"
    >
      <li
        className="nav-cart"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Damian Sobczak
           | Wyloguj
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
